### PR TITLE
EQL: Add implicit ordering on timestamp

### PR DESCRIPTION
**Pull request by costin**
QL: Move Sort base class from SQL to QL

### CODE REVIEW

File: x-pack/plugin/eql/src/test/resources/mapping-nested.json [hunk 1]
```diff
@@ -3,6 +3,9 @@
         "event_type" : {
             "type" : "keyword"
         },
+        "timestamp" : {
+            "type" : "keyword"
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/resources/mapping-nested.json]: `keyword`?

File: x-pack/plugin/eql/src/test/resources/mapping-binary.json [hunk 2]
```diff
@@ -3,6 +3,9 @@
         "event_type" : {
             "type" : "keyword"
         },
+        "timestamp" : {
+            "type" : "keyword"
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/resources/mapping-binary.json]: `keyword`?

**Review by astefan — APPROVED** [review]
LGTM. Left two comments.

**Review by aleksmaus — APPROVED** [review]

File: docs/reference/eql/search.asciidoc [hunk 3]
```diff
@@ -27,7 +27,7 @@ PUT sec_logs/_bulk?refresh
 You can now use the EQL search API to search this index using an EQL query.
 
 The following request searches the `sec_logs` index using the EQL query
-specified in the `query` parameter. The EQL query matches events with an
+specified in the `rule` parameter. The EQL query matches events with an
```
Comment (aleksmaus) [inline docs/reference/eql/search.asciidoc]: something didn't merge right, should be `query`

**Comment by costin** [comment]
@elasticmachine run elasticsearch-ci/2

File: docs/reference/eql/search.asciidoc [hunk 4]
```diff
@@ -74,7 +75,8 @@ The API returns the following response containing the matching event:
             "name": "cmd.exe",
             "path": "C:\\Windows\\System32\\cmd.exe"
           }
-        }
+        },
+        "sort" : [1607339167000]
```
Comment (costin) [inline docs/reference/eql/search.asciidoc]: @jrodewig Not sure whether there's a better way to match the document - essentially the score has been replaced by sort and I'd like to remove both from the result.

Comment (jrodewig) [reply docs/reference/eql/search.asciidoc]: @costin  No worries. As long as the CI tests pass, I would not consider this a blocker. I can work on the additional doc changes as a separate PR.

Comment (jrodewig) [reply docs/reference/eql/search.asciidoc]: Related PR: #53027 53027

**Pull request merged**